src/main/scala/model/world/Reproduction.scala

```diff
@@ -26,19 +27,28 @@ object Reproduction {
    * @param dad another bunny
    * @return the 4 children of the couple, one for each cell of the Punnett's square
    */
-  def generateChildren(mom: Bunny, dad: Bunny): Population = {
+  def generateChildren(mom: Bunny, dad: Bunny, mutations: Option[List[Mutation]] = None): Population = {
     var childrenGenotypes = List.fill(CHILDREN_EACH_COUPLE)(PartialGenotype(Map()))
+
     Genes.values.foreach(gk => {
-      val grandmaMomAllele = mom.genotype(gk).momAllele
-      val grandpaMomAllele = mom.genotype(gk).dadAllele
-      val grandmaDadAllele = dad.genotype(gk).momAllele
-      val grandpaDadAllele = dad.genotype(gk).dadAllele
-      val anotherGene = Random.shuffle(
-        List( Gene(gk, grandmaMomAllele, grandmaDadAllele),
-              Gene(gk, grandpaMomAllele, grandmaDadAllele),
-              Gene(gk, grandmaMomAllele, grandpaDadAllele),
-              Gene(gk, grandpaMomAllele, grandpaDadAllele)))
-      childrenGenotypes = (for (i <- 0 until CHILDREN_EACH_COUPLE) yield childrenGenotypes(i) + anotherGene(i)).toList
+      val grandmaMomAllele = StandardAllele(mom.genotype(gk).momAllele.kind)
+      val grandpaMomAllele = StandardAllele(mom.genotype(gk).dadAllele.kind)
+      val grandmaDadAllele = StandardAllele(dad.genotype(gk).momAllele.kind)
+      val grandpaDadAllele = StandardAllele(dad.genotype(gk).dadAllele.kind)
+      val genesOfReproduction : List[Gene]= List( Gene(gk, grandmaMomAllele, grandmaDadAllele),
+                                                  Gene(gk, grandpaMomAllele, grandmaDadAllele),
+                                                  Gene(gk, grandmaMomAllele, grandpaDadAllele),
+                                                  Gene(gk, grandpaMomAllele, grandpaDadAllele) )
+      val anotherGenes = Random.shuffle(genesOfReproduction)
+
+      mutations match {
+        case None => childrenGenotypes = (for (i <- 0 until CHILDREN_EACH_COUPLE) yield childrenGenotypes(i) + anotherGenes(i)).toList
```

Comment:
Dato che `childrenGenotypes = (for (i <- 0 until CHILDREN_EACH_COUPLE) yield childrenGenotypes(i) + anotherGenes(i)).toList` lo fai in entrambi i casi, lo puoi spostare fuori dal case/match per evitare ripetizione